packages/eas-cli/src/commands/worker/deploy.ts

```diff
@@ -0,0 +1,195 @@
+import chalk from 'chalk';
+import fs from 'node:fs/promises';
+import * as path from 'node:path';
+
+import EasCommand from '../../commandUtils/EasCommand';
+import Log from '../../log';
+import { ora } from '../../ora';
+import { createProgressTracker } from '../../utils/progress';
+import * as WorkerAssets from '../../worker/assets';
+import { getSignedDeploymentUrlAsync } from '../../worker/deployment';
+import { UploadParams, batchUploadAsync, uploadAsync } from '../../worker/upload';
+
+const isDirectory = (directoryPath: string): Promise<boolean> =>
+  fs
+    .stat(directoryPath)
+    .then(stat => stat.isDirectory())
+    .catch(() => false);
+
+export default class WorkerDeploy extends EasCommand {
+  static override description = 'deploy an Expo web build';
+  static override aliases = ['deploy'];
+
+  // TODO(@kitten): Keep command hidden until worker deployments are live
+  static override hidden = true;
+  static override state = 'beta';
+
+  static override flags = {
+    // TODO(@kitten): Allow deployment identifier to be specified
+  };
+
+  static override contextDefinition = {
+    ...this.ContextOptions.ProjectConfig,
+    ...this.ContextOptions.ProjectDir,
+    ...this.ContextOptions.LoggedIn,
+  };
+
+  async runAsync(): Promise<void> {
+    Log.warn('EAS Worker Deployments are in beta and subject to breaking changes.');
+
+    const {
+      privateProjectConfig: { projectId, exp },
+      loggedIn: { graphqlClient },
+      projectDir,
+    } = await this.getContextAsync(WorkerDeploy, {
+      nonInteractive: true,
+    });
+
+    const distPath = path.resolve(projectDir, 'dist');
+    const distClientPath = path.resolve(distPath, 'client');
+    const distServerPath = path.resolve(distPath, 'server');
+    if (!(await isDirectory(distPath))) {
+      throw new Error(
+        `No "dist/" folder found at ${distPath}. Prepare your project for deployment with "npx expo export"`
+      );
```

Comment:
Can we do it automatically for the user if `dist` is not present?

Reply:
That's a great shout!
If you don't mind, I created a ticket for it, since there's several extra steps we'd have to do (make sure the export mode is correctly set up and check for Expo Router)
https://linear.app/expo/issue/ENG-13381/eas-cli-on-missing-dist-folder-offer-export